packages/appsync-modelgen-plugin/src/visitors/appsync-json-metadata-visitor.ts

```diff
@@ -160,7 +160,10 @@ export class AppSyncJSONVisitor<
       models: {},
       enums: {},
       nonModels: {},
-      codegenVersion: this._parsedConfig.codegenVersion || 'unknown',
+      // This is hard-coded for the schema version purpose instead of codegen version
+      // To avoid the failure of validation method checkCodegenSchema in JS Datastore
+      // The hard code is starting from amplify codegen major version 4
```

Comment:
It might be worth explaining why/when this **should** be revved. I.e.,

> This value should be revved using semver rules when any protocol details expressed in the `/models` files change.

Reply:
Maybe more to the point, this comment sounds like, "we're setting this to X because DataStore breaks otherwise." But, the value actually just identifies a protocol version, and that protocol version now deviates from package version as of package version 4.

Reply:
Yes this will be the protocol version instead. The current approach is a temporary workaround. I will mark this as a TODO to update the comments once the protocol between codegen and library is well defined after a detailed design